test/parallel/test-runner-mocking.js

```diff
@@ -446,17 +446,27 @@ test('spy functions don\'t affect the prototype chain', (t) => {
 
   const msg = 'ok';
 
+  const ABeforeMockIsUnchanged = Object.getOwnPropertyDescriptor(A, A.someTask.name);
+  const BBeforeMockIsUnchanged = Object.getOwnPropertyDescriptor(B, B.someTask.name);
+  const CBeforeMockShouldNotHaveDesc = Object.getOwnPropertyDescriptor(C, C.someTask.name);
```

Comment:
these should probably hardcode the name, but otherwise this looks much better thanks :-)

Reply:
niiiiiiice. I don't like hardcoded strings because of refactoring (if you change the function name, wouldn't need to see the test breaking to change all hardcoded strings)

Reply:
In that case, i'd say `const fnName = 'someTask'` - rather than relying on a bunch of runtime `.name` lookups.

Reply:
still, putting the function name as a hard-coded string doesn't help. Why not rely on a .name lookup?

Reply:
because it conveys your intention better than relying on a mutable property of a function. more specifically, it's the property name, which need have no correlation to the function's name.

Reply:
is it required to change or can go this way?

Reply:
There's an implicit rule that all the feedback receives during the review process must be addressed, so either accepted or explicitly dismissed, and as long that there are no one explicitly blocking the PR, you'd be fine. In this case, the change is not required, and you can say "I hear you, but I prefer to keep it as is, thanks for the review" and go on.
FWIW, I have a hard time following how typing out `A.someTask.name` is less hard-coded than `'someTask'`, in either case you have to type out `someTask` and we would need to change all the instances of `A.someTask.name` if the name was changed in a refactor. To me, Jordan is making valid point when he says using `.name` doesn't make a lot of sense when the test should not rely on the `.name` property of the function being mocked (and maybe we could add a test with a method that have a different name for completeness sake btw).

Reply:
Given the class and method name as below:

```js
class A {
    static someTask(msg) {
      return msg;
    }
  }
 ```
Imagine I'm accessing its name as `A.someTask.name`

If I use some refactoring tool such as VSCode, go directly to the class and press F2 to rename the function. All occurrences that use the function as `A.someTask.name` will be changed automatically.

If we use the hard-coded string `someTask` and I refactor the method from the class., I will have to check where is this string all over my code and change it manually.

For the case in this test, I believe it's irrelevant as the class itself is being created right before the assertions and both string or reference approaches for the function would work nicely. 

In summary, I'd keep it as it's now just to avoid spending more time on this PR as it's not a blocking thing

Reply:
That's totally fine, it's not blocking.

Overall tho, I don't think "automated refactoring tools" are that useful in a dynamic language like JS, and I don't think that's a major concern here, especially if the property name is stored in a variable physically near the class definition.

Also, this is tests, not production code, and it's *much* better to have a change like this in test code cause failures everywhere else - that way a human is forced to evaluate more uses of the thing being changed :-)

Reply:
There's also the possibility of using a variable both for the method definition, and for accessing it later. Anyway, the current version is good enough IMO.
```js
const fnName = 'someTask';
class A {
    static [fnName](msg) {
      return msg;
    }
  }
const ABeforeMockIsUnchanged = Object.getOwnPropertyDescriptor(A, fnName);
```